### DISCUSSION TITLE
Getting error on custom sigin page [ API resolved without sending a response . . . ]

### DISCUSSION THREAD

**Discussion by mrkentung** [nextauthjs/next-auth#557] [Help]
*Please refer to the [documentation](https://next-auth.js.org/getting-started/introduction), the [example project](https://github.com/iaincollins/next-auth-example) and existing issues before creating a new issue.*

**Your question**
How to create a custom page, because  following the documentation but get error on console ?

![Screenshot_9](https://user-images.githubusercontent.com/15662229/89600241-c081b080-d88b-11ea-857f-b6f064e1e91b.png)

**What are you trying to do**
i'm trying to crate custom signin page

**Documentation feedback**
*Documentation refers to searching through [online documentation](https://next-auth.js.org), code comments and issue history. The example project refers to [next-auth-example](https://github.com/iaincollins/next-auth-example).*

* [ ] Found the documentation helpful
* [x] Found documentation but was incomplete
* [ ] Could not find relevant documentation
* [ ] Found the example project helpful
* [ ] Did not find the example project helpful

**Comment by iaincollins** [comment]
Hi there! Can you share the code you are using?

**Comment by mrkentung** [comment]
This custom page code, following the [documentation ](https://next-auth.js.org/configuration/pages)

![Screenshot_13](https://user-images.githubusercontent.com/15662229/89698854-aa3c2900-d94d-11ea-95d2-0df7f51a38f5.png)

**Comment by ndom91** [comment]
Are you trying to use the custom credentials or an oauth provider? Because the route your going only works with custom username / password providers. If you're using any of the oauth providers, then you want to take a look [here](https://next-auth.js.org/configuration/pages#oauth-sign-in) or [here](https://next-auth.js.org/configuration/pages#email-sign-in) for Email (magic-link) type signin

**Comment by ziyadedher** [comment]
Sorry for reviving a dead Q&A but this shows up on the front-page of Google when searching for the error.

The problem for me was on the server side: I was not awaiting the call to `NextAuth` in the `[...nextauth].ts`. Awaiting it fixed the problem.

**Comment by andresvo** [comment]
I was getting the same error message, but my mistake was putting my custom signin.js file in pages/api/auth/ instead of pages/auth/
